### DISCUSSION TITLE
Attach <Dropdown.Content /> to a dynamically rendered element

### DISCUSSION THREAD

**Discussion by sndp-s** [radix-ui/primitives#1864] [Help] [answered]
I need to be able to attach my `<DropdownMenu.Content />` to a dynamically rendered element. (an @mention text rendered in a `<span/>` in a (doist-typist) rich text editor).

Right now I have two ways to position the `<DropdownMenu.Content/>` of my `<Dropdown />`

1. `<DropdownMenu.Trigger/>` (`<DropdownMenu.Content/>` gets placed around it)

2. `<DropdownMenu.Portal/>` (Portals the `<DropdownMenu.Content/>` to the element specified in the 'container' prop)

### Approach 1.

DropdownMenu.js

```react.js
<DropdownMenu.Root>
    <DropdownMenu.Trigger />
    <DropdownMenu.Portal>
        <DropdownMenu.Content>
            <MenuItem1/>
            <MenuItem2/>
            <MenuItem3/>
        </DropdownMenu.Content>
    </DropdownMenu.Portal>
</DropdownMenu.Root>
```

![image](https://user-images.githubusercontent.com/23122564/210521729-06a699f6-e231-480e-be29-7ed9d8f64bbb.png)

I see the problem with my code here, The Trigger is in `<DropdownMenu />` while it is supposed to be at the `<span>@</span>` but can I accompolish this? (by wrapping my editor in the `<DropdownMenu.Root>` ? but then again how will I attach the `<DropdownMenu..Trigger />`) to the `<span>@</span>`?

### Approach 2.

I get a prop called `decorationNode` (its the @mention node, `<span>@<span/>` I talked about in the previous point) in my `<DropdownMenu />` component. I use this as the value `container` prop of the `<DropdownMenu.Portal>` and this almost work

Here is the result:

![image](https://user-images.githubusercontent.com/23122564/210533938-c27e2e19-6cb3-48f9-8709-05679d598539.png)

it is almost good in that it is rendered around the correct element but then again it has become a part of the editor and not something attached to it.

Both of these don't seem to fit my use case. Is there an other to accomplish this?

**Comment by joaom00** [comment]
I think a `Dropdown` isn't the right component for you usecase. Maybe `Popover` with `Popover.Anchor`.

**Reply by sndp-s** [reply]
> I think a `Dropdown` isn't the right component for you usecase. Maybe `Popover` with `Popover.Anchor`.

That does seem like it but then again I need to wrap the `Popover.Anchor` around @mention 's `<span>@</span>` which is the part I am unable to figure out. Any idea on that?

**Comment by benoitgrelard** [comment, marked as answer]
Yeah it looks like you're after some kind of combobox/autocomplete component which we currently don't have.
Ariakit has an example close to that using their combobox hooks: https://ariakit.org/examples/combobox-textarea

**Reply by joaom00** [reply]
Is possible to use `Popover` to build something like the ariakit example? I mean, the `Popover.Content` position relative to the position of the caret in input? without the autocomplete stuff

**Reply by benoitgrelard** [reply]
It might be possible, although I'm not sure of the accessibility impact given that it's not directly made for that pattern.

**Reply by joaom00** [reply]
I was thinking of building the ariakit example but using radix + [cmdk](https://github.com/pacocoursey/cmdk) because has API similar to radix

**Reply by sndp-s** [reply]
Thanks
